### PR TITLE
Split CI steps

### LINKED ISSUE #189: Split the CI build between frontend and backend checks
Once https://github.com/wagtail/guide/issues/111 is completed we should look at splitting the CI build between frontend and backend checks.

Right now it looks like we’re running front-end linting tasks last even though they’re one of the fastest checks.

We should also ensure we are using the `nvmrc` as the node version reference - see `node-version-file: '.nvmrc'` & https://github.com/actions/setup-node/blob/main/docs/advanced-usage.md#node-version-file

Example implementations

* https://github.com/torchbox/storybook-django/blob/main/.github/workflows/ci.yml
* https://github.com/wagtail/wagtail-hallo/blob/main/.github/workflows/ci.yml

_Originally posted by @thibaudcolas in https://github.com/wagtail/guide/issues/151#issuecomment-1285157031_

### PR DESCRIPTION
**Pull request by lb-**
- Fixes #189
- Adds titles to each job
- Split out the Python (Django) tests to their own job
- Split backend/frontend linting to their own isolated jobs
- Allow make format & make lint to be run as frontend/backend only versions also

### CODE REVIEW

**Comment by lb-** [comment]
OK this should be good to review now 🤞

File: Makefile [hunk 1]
```diff
@@ -15,13 +15,21 @@ test-coverage:
...
 
 lint:
 	poetry run flake8 apps
 	poetry run isort --check-only --diff apps
 	poetry run black --check --diff apps
+ifeq ($(backend-only),true)
+	echo "skipping frontend linting"
+else
 	npm run lint
+endif
```
Comment (laymonage) [inline Makefile:32]: How about splitting these into `format-backend`, `format-frontend`, `lint-backend`, `lint-frontend`, and then making `format` and `lint` call both the `backend` and `frontend` variants?

Kind of like bakerydemo: https://github.com/wagtail/bakerydemo/blob/main/Makefile

Comment (lb-) [reply Makefile:32]: Yeah we did a similar thing in the Wagtail repo also (split into different make commands), but I recall that being not ideal as it added so many different commands.

Comment (lb-) [reply Makefile:32]: will do this though

**Review by laymonage — APPROVED** [review]

**Comment by allcaps** [comment]
TY! :clap:

**Pull request merged**